src/shaders/MaterialX/funcs.h

```diff
@@ -81,34 +83,52 @@ color fgamma(color in, float g){ return sign(in) * pow(abs(in), g);}
 color premult(color in, float alpha) { return in * alpha; }
 color unpremult(color in, float alpha) { return in * (1/alpha); }
 
-float contrast(float in, float amount, float pivot){
+float contrast(float in, float amount, float pivot)
+{
     float out = in - pivot;
     out *= amount;
     out += pivot;
     return out;
 }
 
-color contrast(color in, color amount, color pivot){
+color contrast(color in, color amount, color pivot)
+{
     color out = in - pivot;
     out *= amount;
     out += pivot;
     return out;
 }
 
-color contrast(color in, float amount, float pivot){
+color contrast(color in, float amount, float pivot)
+{
     color out = in - pivot;
     out *= amount;
     out += pivot;
     return out;
 }
-float exponent(float in, float amount){ return (in > 0) ? pow(in, amount) : in;}
-color exponent(color in, color amount){ return (luminance(in) > 0) ? pow(in, amount) : in; }
-color exponent(color in, float amount){ return (luminance(in) > 0) ? pow(in, amount) : in; }
+float exponent(float in, float amount)
+{ 
+    return (in > 0) ? pow(in, amount) : in;
```

Comment:
I think exponent(in,amount) is just OSL's pow(in,amount). It's already a "safe" version.